ingestion/src/metadata/utils/fqdn_separator.py

```diff
@@ -0,0 +1,16 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from metadata.config.common import FQDN_SEPARATOR
+
+
+def get_fqdn(*args):
```

Comment:
@ulixius9 This requires caller to understand the FQN order i.e in tables case it would be service_name.database.schema.table_name and similarly for chart it would be service_name.dashboard.chart 
if the callers make a mistake in the order of these arguments then we will have a bug. 
So instead of this why don't we pass a named arguments

def table_fqn(service_name: str, database_name: str, schema_name:str, table_name: str):
      return f"{service_name}{FQDN_SEPARATOR}{database_name}{FQDN_SEPARATOR}{schema_name}{FQDN_SEPARATOR}{table_name"

Reply:
@pmbrull can you review it as well

Reply:
Agree with @harshach. We know the proper logic but we should have some helper in the client. We could create a method to `OpenMetadata` python client `get_fqdn(create: C)`, which calls an inner function with a `singledispatch` that knows how to build the FQDN one by one for all entities. E.g.,

```python
from functools import singledispatch


@singledispatch
def _get_fqdn(entity):
    raise NotImplemented("...")


@_get_fqdn.register
def _(entity: CreateTableRequest):
    return FQDN_SEPARATOR.join((entity.service, entity.database, entity.dbSchema, entity.name))
```

WDYT? better to have a bit more control and avoid issues by typing the implementations